### PR TITLE
WIP: update the book

### PR DESCRIPTION
**Pull request by dvberkel**
This pull request is here to track the progress and allow review

### CODE REVIEW

**Comment by dvberkel** [comment]
@badboy I have made some changes in the book. I stopped before the webserver and brain parts. Could you maybe review the changes?

**Comment by badboy** [comment]
I'll take a look

File: src/background/boid.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
 # Boid
 A _boid_ is a virtual creatures, much like a bird or a fish. You should get
```
Comment (badboy) [inline src/background/boid.md]:
```suggestion
A _boid_ is a virtual creature, much like a bird or a fish. You should get
```

File: src/background/boid.md [hunk 2]
```diff
@@ -31,8 +31,16 @@ doesn't have edges. If a boid moves off the right side of the world it enters on
...
+will all lie between 0 and 1. E.g. a boid with an _x_ value of 0 is found at the
+left side of the world, whereas a boid with an _x_ value of 1 will find itself at 
+the right side of the world.
+
+Similarly, an _y_ value of 0 corresponds with the top edge and an _y_ value of 1
+corresponds with the bottom edge.
+
 #### Heading
-An other characteristic of a boid is its _heading_, i.e. The angle it is
+An other primary characteristic of a boid is its _heading_, i.e. The angle it is
```
Comment (badboy) [inline src/background/boid.md]:
```suggestion
Another primary characteristic of a boid is its _heading_, i.e. The angle it is
```

File: src/background/alignment.md [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Alignment
-![Try to fly in the same average heading of your cohort](image/alignment.gif)
+![Try to fly in the same average heading of your cohort](../image/alignment.gif)
 
 Aligning yourself in the average heading of your cohort is an other motivation.
```
Comment (badboy) [inline src/background/alignment.md:4]:
```suggestion
Aligning yourself in the average heading of your cohort is another motivation.
```

File: src/background/others.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+1. Separation
+2. Alignment
+3. Cohesion
+
+But these do not have to be the only motivators.
+
+One could imagine that the boids prefer a certain heading over others,
+because it aligns them with a magnetic field.
+
+Or biods prefer a certain location, because it offers a wide variety of
```
Comment (badboy) [inline src/background/others.md]:
```suggestion
Or boids prefer a certain location, because it offers a wide variety of
```

File: src/server.md [hunk 5]
```diff
@@ -2,3 +2,21 @@
 In this chapter we will describe the web server that you will be creating. If
 you are using a starter kit, most of what is described here is already taken
 care of.
+
+## Communication
+An important part of the workshop is the simulation of multiple flocks of boids.
+This simulation is done by the _workshop server_. It keeps track of the different
+teams participating in the workshop. Each team has it's own flock of boids that
```
Comment (badboy) [inline src/server.md]:
```suggestion
teams participating in the workshop. Each team has its own flock of boids that
```

File: src/server.md [hunk 6]
```diff
@@ -2,3 +2,21 @@
...
+they control.
+
+![Communication between workshop server and brain server](image/communication.png)
+
+Each team will run a server that will determine how their flock behaves. This is the
+_brain server_.
+
+The workshop server will periodically send every teams brain server a request. In this
```
Comment (badboy) [inline src/server.md]:
```suggestion
The workshop server will periodically send every team's brain server a request. In this
```

File: src/server.md [hunk 7]
```diff
@@ -2,3 +2,21 @@
...
+request the brain server will find a representation of their flock. The response that 
+the brain server will give it the intent of each boid. I.e. given this situation, what
```
Comment (badboy) [inline src/server.md]:
```suggestion
the brain server will give is the intent of each boid. I.e. given this situation, what
```

File: src/server/representation.md [hunk 8]
```diff
@@ -1,44 +1,38 @@
...
   }
 }
 ```
 
 ## Output
-Once you have calculated how your boids will behave, you can send your response
-back. It needs to contain the intended heading and speed of each and every boid
-in your flock.
+Once you have calculated the intentions of your boids, you can send your response
+back. The response should map a boids name to its intended heading and speed. You do
```
Comment (badboy) [inline src/server/representation.md]:
```suggestion
back. The response should map a boid's name to its intended heading and speed. You do
```

**Review by badboy — COMMENTED** [review]
this is starting to look very good.
I found a bunch of small typos, but nothing major.

File: src/background/key.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Key Ideas
+The beauty of Reynolds boiding algorithm is that the complex organisation
+of a flock stems from a few local rules.
+
+There is no director instructing each bird what to do, but each bird determines
+for her self what to do. The consideration she is making is balancing the following
+driving factors
+
+1. Seperation
```
Comment (badboy) [inline src/background/key.md]:
```suggestion
1. Separation
```

**Comment by dvberkel** [comment]
Hi @badboy , 
How are things for you? I hope you are safe and healthy. 

I have been writing parts of the book again. I think it needs a chapter or two to be completely finished. Could you take another look?

Next would be polishing the workshop server and some other starter kits

**Comment by badboy** [comment]
I'll try to give it a look

**Pull request merged**